packages/core/src/button/base-button.element.scss

```diff
@@ -82,8 +82,22 @@
   box-shadow: 0 #{$cds-global-space-1} 0 0 var(--box-shadow-color) inset;
 }
 
-:host(:hover) {
-  --background: #{$cds-alias-status-info-shade};
+:host(:hover),
+:host(:focus),
+:host(:active) {
+  .private-host::after {
+    background: var(--background);
+    border-radius: var(--border-radius);
+    filter: brightness(95%);
+    position: absolute;
+    content: '';
+    top: 0;
+    left: 0;
+    bottom: 0;
+    right: 0;
+    inset: 0; // once safari catches up positions can be deleted for inset
+    z-index: -1;
+  }
```

Comment:
This trick I figured out lets us set a darker background color programmatically without darkening the text color preserving the high contrast. Works in all browsers.